src/compiler/checker.ts

```diff
@@ -5594,7 +5613,7 @@ namespace ts {
                         localName,
                         typeParamDecls,
                         heritageClauses,
-                        [...indexSignatures, ...staticMembers, ...constructors, ...members]
+                        [...indexSignatures, ...staticMembers, ...constructors, ...publicProperties, ...privateProperties]

```

Comment:
Name bikeshedding:

It looks like `privateProperties` doesn't include keyword `private` properties, so `privateIdentifierProperties` might be a clearer name.